src/apply.jl

```diff
@@ -123,17 +123,19 @@ end
 _gate_vertices(o::ITensor, ψ) = neighbor_vertices(ψ, o)
 _gate_vertices(o::AbstractEdge, ψ) = [src(o), dst(o)]
 
-function _contract_gate(o::ITensor, ψv1, ψv2)
-  Qᵥ₁, Rᵥ₁ = qr(ψv1, setdiff(uniqueinds(ψv1, ψv2), commoninds(ψv1, o)))
-  Qᵥ₂, Rᵥ₂ = qr(ψv2, setdiff(uniqueinds(ψv2, ψv1), commoninds(ψv2, o)))
-  theta = noprime(Rᵥ₁ * Rᵥ₂ * o)
+function _contract_gate(o::ITensor, ψv1, Λ, ψv2)
+  indsᵥ₁ = noncommoninds(ψv1, Λ)
+  Qᵥ₁, Rᵥ₁ = qr(ψv1, setdiff(uniqueinds(indsᵥ₁, ψv2), commoninds(indsᵥ₁, o)))
+  Qᵥ₂, Rᵥ₂ = qr(ψv2, setdiff(uniqueinds(ψv2, indsᵥ₁), commoninds(ψv2, o)))
+  theta = noprime(noprime(Rᵥ₁ * Λ) * Rᵥ₂ * o)
   return Qᵥ₁, Rᵥ₁, Qᵥ₂, Rᵥ₂, theta
 end
 
-function _contract_gate(o::AbstractEdge, ψv1, ψv2)
-  Qᵥ₁, Rᵥ₁ = qr(ψv1, uniqueinds(ψv1, ψv2))
-  Qᵥ₂, Rᵥ₂ = qr(ψv2, uniqueinds(ψv2, ψv1))
-  theta = Rᵥ₁ * Rᵥ₂
+function _contract_gate(o::AbstractEdge, ψv1, Λ, ψv2)
+  indsᵥ₁ = noncommoninds(ψv1, Λ)
+  Qᵥ₁, Rᵥ₁ = qr(ψv1, uniqueinds(indsᵥ₁, ψv2))
+  Qᵥ₂, Rᵥ₂ = qr(ψv2, uniqueinds(ψv2, indsᵥ₁))
+  theta = noprime(Rᵥ₁ * Λ) * Rᵥ₂
   return Qᵥ₁, Rᵥ₁, Qᵥ₂, Rᵥ₂, theta
 end
```

Comment:
I think this function should be:

``` 
function _contract_gate(o::AbstractEdge, ψv1, Λ, ψv2)
  Qᵥ₁, Rᵥ₁ = qr(ψv1, uniqueinds(ψv1, ψv2))
  Qᵥ₂, Rᵥ₂ = qr(ψv2, uniqueinds(ψv2, ψv1))
  theta = noprime(Rᵥ₁ * Λ) * Rᵥ₂
  return Qᵥ₁, Rᵥ₁, Qᵥ₂, Rᵥ₂, theta
end
```

As the call to `noncommoninds ` puts primed indices in `indsᵥ₁` that shouldn't be there?